apps/bixarena/app/bixarena_app/page/bixarena_battle.py

```diff
@@ -39,22 +42,48 @@
 anony_names = ["", ""]
 models = []
 
-# Global state for prompt examples
-prompt_manager = get_prompt_manager()
 
+def example_prompt_cards(num_prompts=3):
+    """Create prompt example cards"""
+    try:
+        # Configure the API client
+        configuration = Configuration(host="http://bixarena-api:8112/v1")
 
-def create_suggested_prompts(num_prompts=3, max_chars=240):
-    """Create suggested prompts cards"""
-    display_prompts = prompt_manager.get_display_prompts(num_prompts, max_chars)
+        # Create API client and example prompt API instance
+        with ApiClient(configuration) as api_client:
+            api_instance = ExamplePromptApi(api_client)
 
-    prompt_cards = []
-    for display_text, full_prompt in display_prompts:
-        btn = gr.Button(
-            value=display_text,
-            elem_classes=["suggested-prompt-card"],
-        )
-        prompt_cards.append((btn, full_prompt))
+            # Create search query to get active prompts only
+            search_query = ExamplePromptSearchQuery(active=True)
 
+            # Fetch example prompts
+            response = api_instance.list_example_prompts(
+                example_prompt_search_query=search_query
+            )
+
+            # Extract questions and sample randomly
+            display_prompts = random.sample(
+                [prompt.question for prompt in response.example_prompts],
+                min(num_prompts, response.total_elements),
+            )
```

Comment:
The behavior of this feature is not fully define here because the page size is not specified. Here, the behavior of the feature would change if the default page size changes in the future. The feature would even break if this number become smaller than `num_prompts`.

Always try to identify all the variables that may affect the behavior of a feature, assume that most of them may change in the future, and add guardrails to ensure that these changes don't go unnoticed.

For example, if the default page size was reduced to `num_prompts`, the app wouldn't not beak but the feature would as the same three prompts would always be returned (deterministic prompts).

The more variables you can control and fix, the more robust the feature will be.